src/infra/services/servicesQueryRepo.go

```diff
@@ -586,67 +619,165 @@ func (repo *ServicesQueryRepo) installableServiceFactory(
 }
 
-func (repo *ServicesQueryRepo) ReadInstallables() ([]entity.InstallableService, error) {
-	installableServices := []entity.InstallableService{}
+func (repo *ServicesQueryRepo) ReadInstallableItems(
+	requestDto dto.ReadInstallableServicesItemsRequest,
+) (installableItemsDto dto.ReadInstallableServicesItemsResponse, err error) {
+	_, err = os.Stat(infraEnvs.ServiceInstalledItemsDir)
+	if err != nil {
+		servicesCmdRepo := NewServicesCmdRepo(repo.persistentDbSvc)
+		err = servicesCmdRepo.RefreshInstallableItems()
+		if err != nil {
+			return installableItemsDto, errors.New(
+				"RefreshServiceInstallableItemsError: " + err.Error(),
+			)
+		}
+	}
 
-	serviceFiles, err := fs.ReadDir(assets, "assets")
+	rawInstallableFilesList, err := infraHelper.RunCmdWithSubShell(
+		"find " + infraEnvs.ServiceInstalledItemsDir + " -type f " +
+			"\\( -name '*.json' -o -name '*.yaml' -o -name '*.yml' \\) " +
+			"-not -path '*/.*' -not -name '.*'",
+	)
 	if err != nil {
-		return installableServices, errors.New("ReadServiceFilesError: " + err.Error())
+		return installableItemsDto, errors.New(
+			"ReadInstallableFilesError: " + err.Error(),
+		)
 	}
 
-	for _, serviceFile := range serviceFiles {
-		serviceFileName := serviceFile.Name()
-		rawServiceFilePath := "assets/" + serviceFileName
-		serviceFilePath, err := valueObject.NewUnixFilePath(rawServiceFilePath)
+	if len(rawInstallableFilesList) == 0 {
+		return installableItemsDto, errors.New("NoInstallableFilesFound")
+	}
+
+	rawInstallableFilesListParts := strings.Split(rawInstallableFilesList, "\n")
+	if len(rawInstallableFilesListParts) == 0 {
+		return installableItemsDto, errors.New("NoInstallableFilesFound")
+	}
+
+	installableServices := []entity.InstallableService{}
+	for _, rawFilePath := range rawInstallableFilesListParts {
+		itemFilePath, err := valueObject.NewUnixFilePath(rawFilePath)
 		if err != nil {
-			slog.Debug(
-				"InvalidServiceFilePathError",
-				slog.Any("assetFile", rawServiceFilePath),
-			)
+			slog.Debug(err.Error(), slog.String("filePath", rawFilePath))
 			continue
 		}
-		serviceFilePathStr := serviceFilePath.String()
 
-		installableService, err := repo.installableServiceFactory(serviceFilePath)
+		installableService, err := repo.installableServiceFactory(itemFilePath)
 		if err != nil {
 			slog.Debug(
-				"InstallableServiceFactoryError",
-				slog.String("assetFile", serviceFilePathStr),
-				slog.Any("error", err),
+				"CatalogMarketplaceItemFactoryError",
+				slog.String("filePath", itemFilePath.String()), slog.Any("err", err),
 			)
 			continue
 		}
 
 		installableServices = append(installableServices, installableService)
 	}
 
-	return installableServices, nil
+	filteredInstallableServices := []entity.InstallableService{}
+	for _, installableService := range installableServices {
+		if requestDto.ServiceName != nil {
+			isNameEqual := strings.EqualFold(
+				installableService.Name.String(), requestDto.ServiceName.String(),
+			)
+			if !isNameEqual {
```

Comment:
What's the benefit of using EqualFold instead of simple comparison?

Reply:
I considered the likelihood of users entering names in uppercase or with varying letter cases, which is a common typo. So instead of worrying about calling `ToLower` in each case where I need to compare strings, I decided to use [`EqualFold`](https://pkg.go.dev/strings#EqualFold). This function handles the case-insensitive comparison for me more comprehensively.

Additionally, I ran several performance tests, and the impact of this function on execution speed is negligible.